api-get-policy.go

```diff
@@ -35,7 +35,8 @@ func (c Client) GetBucketPolicy(bucketName, objectPrefix string) (bucketPolicy p
 		return policy.BucketPolicyNone, err
 	}
 	policyInfo, err := c.getBucketPolicy(bucketName)
-	if err != nil {
+	errResponse := ToErrorResponse(err)
+	if err != nil && errResponse.Code != "NoSuchBucketPolicy" {
```

Comment:
GetBucketPolicy is overridden Core API @vadmeste, what you are talking about is GetBucketPolicy(bucket, prefix) which is not changed and should be working as expected.